src/chan_vese.jl

```diff
@@ -0,0 +1,258 @@
+"""
+    chan_vese(img; μ, λ₁, λ₂, tol, max_iter, Δt, reinitial_flag)
+
+Segments image `img` by evolving a level set. An active contour model 
+which can be used to segment objects without clearly defined boundaries.
+
+# output
+Return a `BitMatrix`.
+
+# Details
+
+Chan-Vese algorithm deals quite well even with images which are quite
+difficult to segment. Since CV algorithm relies on global properties, 
+rather than just taking local properties under consideration, such as
+gradient. Better robustness for noise is one of the main advantages of 
+this algorithm. See [1], [2], [3] for more details.
+
+# Options
+
+The function argument is described in detail below. 
+
+Denote the edge set curve with 𝐶 in the following part.
+
+## `μ::Float64`
+
+The argument `μ` is a weight controlling the penalty on the total length
+of the curve 𝐶;
+
+For example, if the boundaries of the image are quite smooth, a larger `μ`
+can prevent 𝐶 from being a complex curve.
+
+Default: 0.25
+
+## `λ₁::Float64`, `λ₂::Float64`
+
+The argument `λ₁` and `λ₂` affect the desired uniformity inside 𝐶 and 
+outside 𝐶, respectively. 
+
+For example, if set `λ₁` < `λ₂`, we are more possible to get result with 
+quite uniform background and varying grayscale objects in the foreground.
+
+Default: λ₁ = 1.0
+         λ₂ = 1.0
+
+## `tol::Float64`
+
+The argument `tol` controls the level set variation tolerance between 
+iteration. If the L2 norm difference between two level sets of adjacent
+iterations is below `tol`, then the solution will be assumed to be reached.
+
+Default: 1e-3
+
+## `max_iter::Int64`
+
+The argument `max_iter` controls the maximum of iteration number.
+
+Default: 500
+
+## `Δt::Float64`
+
+The argument `Δt` is a multiplication factor applied at calculations 
+for each step, serves to accelerate the algorithm. Although larger `Δt`
+can speed up the algorithm, it might prevent algorithm from converging to 
+the solution.
+
+Default: 0.5
+
+## reinitial_flag::Bool
+
+The arguement `reinitial_flag` controls whether to reinitialize the
+level set in each step.
+
+Default: false
+
+# Examples
+
+```julia
+using TestImages
+using ImageSegmentation
+
+img = testimage("cameraman")
+
+cv_result = chan_vese(img, μ=0.25, λ₁=1.0, λ₂=1.0, tol=1e-3, max_iter=200, Δt=0.5, reinitial_flag=false)
```

Comment:
No need to explicit write default values in the example:

```suggestion
cv_result = chan_vese(img, max_iter=200)
```